app/survey_config/survey_config.py

```diff
@@ -61,3 +63,11 @@ def get_service_links(  # pylint: disable=unused-argument, no-self-use
         ru_ref: Optional[str],
     ) -> Optional[list[dict]]:
         return None
+
+    def get_data_layer(  # pylint: disable=no-self-use
+        self, tx_id: Optional[str] = None
+    ) -> Optional[list[dict]]:
```

Comment:
This should always be returning a list unless you are explicitly checking for None, which it doesn't seem to do at the moment.

Reply:
Fixed this type hint now thanks!